### PR TITLE
Add config to disable automatic assignment assignee reassignment [SDCP-860]

### PR DESCRIPTION
**Pull request by devketanpro**
This PR introduces a new configuration, `ASSIGNMENT_MANUAL_REASSIGNMENT_ONLY`, to allow manual control over assignment reassignments. When enabled, actions like saving or moving an article will no longer automatically update the assignee. Manual reassignment becomes the only way to change the assignee

### CODE REVIEW

File: server/planning/assignments/assignments.py [hunk 1]
```diff
@@ -862,7 +863,9 @@ def _get_empty_updates_for_assignment(self, assignment):
 
     def _set_user_for_assignment(self, assignment, assignee, assignor=None):
         updates = self._get_empty_updates_for_assignment(assignment)
-        updates["assigned_to"]["user"] = assignee
+
+        if not get_config_assignment_manual_reassignment_only():
+            updates["assigned_to"]["user"] = assignee
 
         if assignor:
```
Comment (MarkLark86) [inline server/planning/assignments/assignments.py]: If the config is turned off, then we should probably not change the `assignor_user` either (as no re-assignment was performed)

**Review by MarkLark86 — COMMENTED** [review]
Would be good to have some tests too, for when this config is turned ON

File: server/planning/assignments/assignments.py (hunk 1 above)
Comment (devketanpro) [reply server/planning/assignments/assignments.py]: got it

**Review by MarkLark86 — COMMENTED** [review]
Looks good, but can you please investigate the failing tests.

**Comment by devketanpro** [comment]
I checked on the develop branch, and the tests are failing there as well. It seems the issue is not related to these changes.I'll continue investigating further.

**Comment by devketanpro** [comment]
Hey @MarkLark86, Here is the PR for the test case fix: https://github.com/superdesk/superdesk-planning/pull/2138

**Review by MarkLark86 — APPROVED** [review]

**Pull request merged**